bevy_lint/src/lints/insert_event_resource.rs

```diff
@@ -0,0 +1,226 @@
+//! Checks for the `Events<T>` resource being manually inserted through `App::init_resource()` or
+//! `App::insert_resource()` instead of with `App::add_event()`.
+//!
+//! # Why is this bad?
+//!
+//! While `Events<T>` is technically a `Resource`, you cannot register an event by _just_ inserting
```

Comment:
Somebody who writes this code unintentionally will also not know what it means to "register" an event or why that matters, will not be familiar with the `EventRegistry` or why it exists, and will not know what the "desired behavior" is. So this whole paragraph isn't as helpful as it could be while also tying it to the current codebase's implementation.

We should either be much more specific about what doesn't happen (not the mechanics of *how* it happens) when you do this (for example: what it means to "update" an event and that that only happens when the event is registered with an `EventRegistry`), or we should be vague enough to not have to rewrite this message if behavior or naming changes, and instead lean on "go read `App::add_event` if you're curious about what else is initialized/handled by `add_event`".

"Unless you have intentionally and knowingly initialized the `Events<T>` `Resource` in this way, events and their resources should be initialized with `add_event`, which will initialize the `Events<T>` `Resource` and more."

Reply:
Some very good points, thank you! I took a step back and tested why manually inserting `Events<T>` is bad: it can result in a memory leak if you don't call `Events::update()`. `App::add_events()` automatically registers a system that does this, which is why it's preferred.

I rephrased everything in 28aba235202b48e10ceca23d0356b55b1d881f07, using your suggestion as a starter. Thanks!